### PR TITLE
Relax AR version, add test for different version

### PR DESCRIPTION
**Pull request by palkan**
Thanks for the gem!

I'm using it with Rails 6 – it works) So, let's relax the constraints.

Added travis configuration to run tests against different versions (to make sure that it really works).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,12 +4,38 @@ language: ruby
...
   - 2.5.1
+
+addons:
+  postgresql: "9.6"
+
 services:
-  - postgresql
   - mysql
+
+before_install:
```
Comment (palkan) [inline .travis.yml]: See https://docs.travis-ci.com/user/languages/ruby/#bundler-20

File: database_validations.gemspec [hunk 2]
```diff
@@ -30,5 +30,5 @@ and ActiveRecord validations with better performance and consistency."
   spec.add_development_dependency 'rspec', '~> 3.0'
   spec.add_development_dependency 'rubocop', '~> 0.60'
   spec.add_development_dependency 'rubocop-rspec', '~> 1.30'
-  spec.add_development_dependency 'sqlite3', '~> 1.3'
+  spec.add_development_dependency 'sqlite3', '~> 1.3.0'
```
Comment (palkan) [inline database_validations.gemspec]: See https://github.com/rails/rails/pull/35154

File: .travis.yml [hunk 3]
```diff
@@ -4,12 +4,38 @@ language: ruby
 cache: bundler
 rvm:
   - 2.5.1
+
+addons:
+  postgresql: "9.6"
```
Comment (palkan) [inline .travis.yml]: Travis default version is outdated (9.0), Rails 6 requires at least 9.3

**Comment by palkan** [comment]
AR 3.2 is not supported by this gem: https://travis-ci.org/toptal/database_validations/jobs/496140442

Ruby <2.5 is not supported, too: https://travis-ci.org/toptal/database_validations/jobs/496143426 (so this one, probably, should be easy to fix).

**Comment by pirj** [comment]
Except probably the gem is only compatible with AR 4.2+.

**Comment by palkan** [comment]
> Except probably the gem is only compatible with AR 4.2+.

Yeah, it should be. I'm trying to make specs work)

**Comment by palkan** [comment]
@pirj What about Ruby 2.4? We can try to make it work, too

**Comment by palkan** [comment]
> the gem is only compatible with AR 4.2+.

It's not( https://travis-ci.org/toptal/database_validations/jobs/496152914

**Comment by pirj** [comment]
@palkan Definitely, I believe something like `klass.__send__(:alias_method` will fix the issue.

As per [this writeup](https://semaphoreci.com/blog/2017/11/08/ruby-versions-used-in-commercial-projects-in-2017.html), which is only a bit over one year old, about a third of installations use Rubies that don't receive security updates anymore. Keeping in mind that 2.3 will fall into this category by April, it doesn't make much sense to support versions prior to 2.4.

**Comment by pirj** [comment]
>> the gem is only compatible with AR 4.2+.
> It's not

You're right!
`Unknown key: :optional` should be relatively easy to fix as well by picking either `options.merge(optional: true)` or `options.merge(required: false)` depending on AR version.

`undefined method `details' for #<ActiveModel::Errors` might be more tricky. Let's probably leave it up for @djezzzl to add support for 4.2?

**Comment by palkan** [comment]
Finally, we're green)

Not supported by Rails 4 features are excluded from specs (using `RAILS_5` constant and a few changes).

Differences in `belongs_to` are handled in code (`optional` vs `required` and error messages).

Let's try to make it work on Ruby 2.4

**Comment by palkan** [comment]
Ok, Ruby 2.4 is fixed, too

**Comment by djezzzl** [comment]
Thank you for your contribution!

**Comment by pirj** [comment]
👏

**Pull request merged**